app/Http/Controllers/PokerController.php

```diff
@@ -1,133 +1,133 @@
-<?php
+    <?php
 
-namespace App\Http\Controllers;
+    namespace App\Http\Controllers;
 
-use Illuminate\Http\Request;
-use Illuminate\Support\Facades\DB;
-use Illuminate\Support\Facades\Broadcast;
-use Illuminate\Support\Facades\Log;
+    use Illuminate\Http\Request;
+    use Illuminate\Support\Facades\DB;
+    use Illuminate\Support\Facades\Broadcast;
+    use Illuminate\Support\Facades\Log;
 
-class PokerController extends Controller
-{
+    class PokerController extends Controller
+    {
 
-    public function createRoom(Request $request){
+        public function createRoom(Request $request){
 
-        $room_id = $this->generateRandomString(); //todo make sure this is unique in db        
+            $room_id = $this->generateRandomString(); //todo make sure this is unique in db        
 
-        $now = date('Y-m-d H:i:s');
-        $id = DB::table('poker_rooms')->insertGetId([
-            'room_id' => $room_id,
-            'password' => $request->password,
-            'is_open' => true,
-            'created_at' => $now,
-            'updated_at' => $now
-        ]);
-
-        $one_week_ago = date("Y-m-d H:i:s", strtotime("-1 week"));
+            $now = date('Y-m-d H:i:s');
+            $id = DB::table('poker_rooms')->insertGetId([
+                'room_id' => $room_id,
+                'password' => $request->password,
+                'is_open' => true,
+                'created_at' => $now,
+                'updated_at' => $now
+            ]);
 
-        DB::table('poker_rooms')->where('created_at', '<=', $one_week_ago)->update([
-            'is_open' => 0
-        ]);
-        return response()->json(['success' => true, 'room_id' => $room_id]);
-    }
+            $one_week_ago = date("Y-m-d H:i:s", strtotime("-1 week"));
```

Comment:
since the $now variable is initialized once per function (and each function can only be executed upon client request) effectively the $now instance is when the request was sent. Creating a global $now variable at the top of the file would not change this since every request would call the controller and reinitialize the global variable. it would however reduce the repeated initializations in each function, and set the initialization in one location which is definitely better.

That said, i dont see how a global value would resolve the edge case you mentioned. Do you have any other recommendations?

Reply:
For this specific function for example, there are 2 calls:

1. call to date() at TimeStamp1, date() = TimeStamp1
2. some DB operation at TimeStamp2
3. call to date() again at TimeStamp3, date() = TimeStamp3

If 2. takes some time to fulfill cause of latency, error, throttling etc. (im assuming its synchronous), then TimeStamp1 < TimeStamp3

If date is initialized once per request, then it would look like this:

1. call to date() at initialization of request at TimeStamp1
2. read date at TimeStamp1, date() = TimeStamp1
3. some DB operation at TimeStamp2
4. read date() again at TimeStamp3, date() = TimeStamp1

In that case regardless of any delays, the scenario described will resolve correctly

Reply:
Now i see what you mean. The issue would be resolved if $now and $one_week_ago were initialized at the same time (or right after each other would be good enough as well)